src/java/org/openzal/zal/MailboxListenerWrapper.java

```diff
@@ -0,0 +1,265 @@
+package org.openzal.zal;
+
+import com.zimbra.cs.mailbox.MailItem;
+import com.zimbra.cs.mailbox.MailboxOperation;
+import com.zimbra.cs.session.PendingModifications;
+import java.util.concurrent.atomic.AtomicBoolean;
+import org.openzal.zal.lib.Clock;
+import org.openzal.zal.log.ZimbraLog;
+import org.openzal.zal.redolog.Redolog;
+
+public abstract class MailboxListenerWrapper
+{
+  public class MailboxListener extends com.zimbra.cs.mailbox.MailboxListener
+  {
+    private final AtomicBoolean isRegistered;
+    private final Clock         mClock;
+
+    private MailboxListener(Clock clock)
+    {
+      mClock = clock;
+      isRegistered = new AtomicBoolean(false);
+    }
+
+    private boolean toBackup(Object what)
+    {
+      byte type = Item.TYPE_UNKNOWN;
+
+      if( what instanceof MailItem.Type )
+      {
+        type = ((MailItem.Type) what).toByte();
+      }
+
+      if( what instanceof MailItem )
+      {
+        Item item = new Item(what);
+        type = item.getType();
+      }
+
+      boolean isRelevant = true;
+      switch( type )
+      {
+        case Item.TYPE_MOUNTPOINT:
+        case Item.TYPE_VIRTUAL_CONVERSATION:
+          isRelevant = false;
+      }
+
+      return isRelevant;
+    }
+
+    @Override
+    public void notify(ChangeNotification notification)
+    {
+     /* $if ZimbraX == 1 $
+      return;
+     /* $else $ */
+      if( isRegistered.get() && notification.mods.hasNotifications() )
+      {
+        MailboxOperation op = notification.op;
+        Operation operation = Operation.SKIP;
+        Account account = new Account(notification.mailboxAccount);
+
+        switch( op.getCode() )
+        {
+          // to be managed with MailboxManager$Listener
+          case Redolog.OP_CREATE_MAILBOX:
+          case Redolog.OP_DELETE_MAILBOX:
+            break;
+          case Redolog.OP_CREATE_SAVED_SEARCH:
+          case Redolog.OP_MODIFY_SAVED_SEARCH:
+          case Redolog.OP_CREATE_TAG:
+          case Redolog.OP_RENAME_TAG:
+          case Redolog.OP_CREATE_CONTACT:
+          case Redolog.OP_MODIFY_CONTACT:
+          case Redolog.OP_EDIT_NOTE:
+          case Redolog.OP_REPOSITION_NOTE:
+          case Redolog.OP_CREATE_NOTE:
+          case Redolog.OP_CREATE_LINK:
+          case Redolog.OP_MODIFY_INVITE_PARTSTAT:
+          case Redolog.OP_CREATE_INVITE:
+          case Redolog.OP_FIX_CALENDAR_ITEM_TIME_ZONE:
+          case Redolog.OP_SET_CALENDAR_ITEM:
+          case Redolog.OP_RENAME_MAILBOX: // TODO: Find out what that operations mean
```

Comment:
rename mailbox is when the comment which contains the user email address is changed in database, could be a useful notification for an account scan